igvm/hypervisor_preferences.py

```diff
@@ -179,25 +178,23 @@ class to let them be prepared lazily.  When Python needs to compare
     executed for that hypervisor by LazyCompare.
 
     The LazyCompare optimization has one other benefit of bring more
-    visibility about selection.  After the sorting is done, we can check
+    visibility about the selection.  After the sorting is done, we can check
     which preferences are actually executed for the selected hypervisor,
-    and log which preference caused this hypervisor to be sorted before
-    the next one.
+    and log which preference caused this hypervisor to be sorted after
+    the previous or before the next one.
     """
-    log.debug('Sorting hypervisors by preference...')
+    plen = len(preferences)
+    log.debug('Sorting hypervisors by {} preferences...'.format(plen))
 
-    # We use decorate-sort-undecorate pattern to log details about sorting.
+    # Use decorate-sort-undecorate pattern to log details about sorting
     for comparables, hypervisor in sorted(
         ([LazyCompare(p, vm, h) for p in preferences], h)
         for h in hypervisors
     ):
-        for index, comparable in enumerate(comparables):
-            if not comparable.executed:
-                break
-        log.info(
-            'Hypervisor "{}" selected with decisive preference {!r} '
-            'after checking {} preferences.'
-            .format(hypervisor, comparable.func, index)
-        )
+        log.info('Hypervisor "{}" selected using {} preferences.'.format(
+            hypervisor, next(
+                (i for i, c in enumerate(comparables) if not c.executed), plen
+            )
+        ))
```

Comment:
Hm, good catch, though I think this makes it much harder to read. I'd prefer we keep the old loop style.
We can even show the correct preference via comparables[index-1], right?

Reply:
Reverted the loop.

We can still show the preference, but it doesn't mean much, because it is not the one that has necessarily caused this hypervisor to be sorted before the next one in line.